### DISCUSSION TITLE
Question concerning the receive() function in FundMe contract

### DISCUSSION THREAD

**Discussion by owanemi** [Cyfrin/foundry-full-course-cu#2259] [Q&A] [answered]
Patrick said we can still fund our fund-me contract without calling the function directly. So he did
```
receive() external payable {
     fund();
 }
 ```
 but then i asked chat gpt the limitations to using the receive function and it said it has a max gas stipend of 2,300 but our fund function which is placed inside "receive()" costs 115,000 gas and if we do a regular transfer, the logic still works, how come?

**Comment by EngrPips** [comment, marked as answer]
Hello @owanemi, I don't think the `receive` function explicitly has a cap on the amount of gas it can use, but what I know is that using `transfer` or `send` will forward only `2300` gas, and the low-level method we use when we trigger the `receive` function of the contract is `call` which forwards almost all the gas available. `receive` function gets triggered whenever someone sends a `call` to our contract which is a call that only sends our contract some value and the `fallback` function is triggered when someone sends a call to our contract which contains some calldata.
